lib/Pool.js

```diff
@@ -155,7 +155,9 @@ class Pool {
         })
         .then(() => {
           // Now we can destroy all the freed resources.
-          this.free.forEach(free => this._destroy(free.resource));
+          return Promise.all(this.free.map(free => this._destroy(free.resource)));
```

Comment:
I understand the point of this, but I think compared to how Tarn used to handle this, it is now potentially flawed since:

1. There's no timeout set for destroy, it could hang forever in worst case
2. It doesn't clear current state after potential error occurs which it did before

This could potentially leave the tarn instance in a state where it has 'tried' to destroy all active resources - but failed - and there's no way to create new resources because it is filled with faulty resources.

I could be thinking too far into it and may be mistaken, but this is my interpretation of this change..

Reply:
Regarding 1., how about we use https://github.com/sindresorhus/p-timeout ?
Regarding 2., sorry I don't understand - how/where does it currently clear state that it doesn't anymore with these changes?

Reply:
Also `Promise.all`  rejects right away if one of the destroy commands fail. So there might be 200 destroys running at the same time, first one rejects and `Promise.all` rejects right away, even that there are 199 promises still unresolved. (see for example this https://stackoverflow.com/questions/31424561/wait-until-all-es6-promises-complete-even-rejected-promises)

Reply:
> Regarding 1., how about we use https://github.com/sindresorhus/p-timeout ?

Since tarn.js actually does have also own timeout for acquire and create already, it would make sense to have it for destroy. No need to include any additional deps for it.

> Regarding 2., sorry I don't understand - how/where does it currently clear state that it doesn't anymore with these changes?

Since _destroy never rejects, I think that functionality is still the same. Anyways reflect should be added there just to be on safe side for future, so that rejected promises will not cause Promise.all to reject too early.

Reply:
Added that as well as an `destroyTimeoutMillis` option